File: Decorator Pattern/Cpp/Decorator/Mocha.cpp
#pragma once

#include "CondimentDecorator.cpp"

class Mocha : public CondimentDecorator {
public:
	Mocha(Beverage* beverage) {
		this->beverage = beverage;
	}

	string getDescription() {
		return beverage->getDescription() + " + Mocha";;
	}

	double cost() {
		return beverage->cost() + 0.20;
	}
};

File: Patterns/Decorator Pattern/CSharp/Decorator/Mocha.cs
using Beverage = Coffee.Beverage;

namespace Decorator 
{
    public class Mocha : CondimentDecorator 
    {
        public Mocha(Beverage? beverage)
        {
            this.beverage = beverage;
        }

        public override string getDescription() => beverage?.getDescription() + " + Mocha";

        public override double cost() => beverage == null ? 0.20 : beverage.cost() + 0.20;
    }
}